examples/component-testing/app/routes/users/$userId.test.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+import { vi } from "vitest";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { useLoaderData } from "remix";
+import UserScreen from "./$userId";
+
+vi.mock("remix", () => {
+  // mock whatever remix APIs you're using
+  return {
+    Form: (props: React.RefAttributes<HTMLFormElement>) => <form {...props} />,
+    useLoaderData: vi.fn(),
+    useLocation: vi.fn(() => ({ key: Math.random().toString(32).slice(2) }))
+  };
+});
```

Comment:
Tbh, I'm also not sure if we should mock it this way 🤔

I think a custom `render` function that takes care of the contexts and has overridable `action`/`links`/`loader`/`meta` would give the team more control on the whole flow and on how things will be tested/mocked

Reply:
I agree, it's just a huge challenge to mock the remix context with something sensible. But I do plan on exploring that approach.